STYLE_GUIDE.md

```diff
@@ -194,12 +218,112 @@ from composer.path.to.module import my_file as my_file
 ```
 
 
-## 6. Hparams and Config Classes
+## 7. Documentation
+
+Composer uses [Google Style Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+
+The following guidelines apply to documentation.
+1.  Each function that needs a docstring must have its input arguments and return statement (if not None) annotated.
+1.  Each argument annotation should include the type. If the argument has a default value, the type annotation should
+    specify "optional", and the docstring should say the default value. Some examples:
+
+    ```python
+
+    def foo(bar: int):
+        """Foo.
+
+        Args:
+            bar (int): Required bar.
+        """
+        ...
+
+    def foo2(bar: int = 42):
+        """Foo2.
+
+        Args:
+            bar (int, optional): Optional bar. (default: ``42``)
```

Comment:
If we're going with
```...bar. Default: ``42`` ```
as the default value format, should show that here (and in below docstrings also)

Reply:
Also I don't think we want people starting off with the word Optional in their variable descriptions (which the examples sort of suggest right now), so maybe change to something like:

```bar (int, optional): The first argument. Default: ``42`` ```

that can't as easily be read as suggesting particular free text content.